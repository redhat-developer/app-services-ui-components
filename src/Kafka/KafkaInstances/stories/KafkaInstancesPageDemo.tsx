import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
