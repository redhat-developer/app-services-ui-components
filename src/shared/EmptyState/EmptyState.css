.pf-c-empty-state.pf-m-xl {
  --pf-c-empty-state__content--MaxWidth: 624px;
}
