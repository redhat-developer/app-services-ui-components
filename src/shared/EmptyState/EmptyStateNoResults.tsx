import {
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title,
} from "@patternfly/react-core";
import { SearchIcon } from "@patternfly/react-icons";
import type { VoidFunctionComponent } from "react";
import { useTranslation } from "react-i18next";

export const EmptyStateNoResults: VoidFunctionComponent = () => {
  const { t } = useTranslation();

  return (
    <EmptyState variant={EmptyStateVariant.large}>
      <EmptyStateIcon icon={SearchIcon} />
      <Title headingLevel="h2" size="lg">
        {t("common:empty_state_no_results_title")}
      </Title>
      <EmptyStateBody>{t("common:empty_state_no_results_body")}</EmptyStateBody>
    </EmptyState>
  );
};
