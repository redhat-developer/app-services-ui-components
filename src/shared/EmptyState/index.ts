export * from "./EmptyState";

/**
 * @deprecated
 */
export {
  EmptyState as AppServicesEmptyState,
  EmptyStateVariant as AppServicesEmptyStateVariant,
} from "./EmptyState";
