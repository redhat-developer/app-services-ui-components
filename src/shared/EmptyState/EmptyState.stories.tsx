import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { EmptyState, EmptyStateVariant } from "./EmptyState";

export default {
  component: EmptyState,
  parameters: {},
} as ComponentMeta<typeof EmptyState>;

const Template: ComponentStory<typeof EmptyState> = (args, { parameters }) => (
  <div style={{ height: parameters.previewHeight as number }}>
    <EmptyState {...args} />
  </div>
);

export const DefaultStory = Template.bind({});
DefaultStory.args = {
  titleProps: {
    title: "Default Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
DefaultStory.storyName = "Default";

export const GettingStartedStory = Template.bind({});
GettingStartedStory.args = {
  emptyStateProps: {
    variant: EmptyStateVariant.GettingStarted,
  },
  titleProps: {
    title: "Getting Started Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
GettingStartedStory.storyName = "Getting Started";

export const NoAccessStory = Template.bind({});
NoAccessStory.args = {
  emptyStateProps: {
    variant: EmptyStateVariant.NoAccess,
  },
  titleProps: {
    title: "No Access Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
NoAccessStory.storyName = "No Access";

export const NoItemsStory = Template.bind({});
NoItemsStory.args = {
  emptyStateProps: {
    variant: EmptyStateVariant.NoItems,
  },
  titleProps: {
    title: "No Items Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
NoItemsStory.storyName = "No Items";

export const NoResultStory = Template.bind({});
NoResultStory.args = {
  emptyStateProps: {
    variant: EmptyStateVariant.NoResult,
  },
  titleProps: {
    title: "No Result Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
NoResultStory.storyName = "No Result";

export const UnexpectedErrorStory = Template.bind({});
UnexpectedErrorStory.args = {
  emptyStateProps: {
    variant: EmptyStateVariant.UnexpectedError,
  },
  titleProps: {
    title: "Unexpected Error Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
UnexpectedErrorStory.storyName = "Unexpected Error";

export const PageNotFoundStory = Template.bind({});
PageNotFoundStory.args = {
  emptyStateProps: {
    variant: EmptyStateVariant.PageNotFound,
  },
  titleProps: {
    title: "Page Not Found Empty State",
    headingLevel: "h1",
  },
  emptyStateBodyProps: {
    body: `
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.`,
  },
};
PageNotFoundStory.storyName = "Page Not Found";
