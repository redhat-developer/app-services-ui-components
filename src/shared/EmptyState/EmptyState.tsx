import type {
  ButtonProps,
  EmptyStateBodyProps,
  EmptyStateIconProps,
  EmptyStateProps as PFEmptyStateProps,
  TitleProps,
} from "@patternfly/react-core";
import {
  Button,
  ButtonVariant,
  EmptyState as PFEmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant as PFEmptyStateVariant,
  Title,
  TitleSizes,
} from "@patternfly/react-core";
import {
  ExclamationCircleIcon,
  LockIcon,
  PlusCircleIcon,
  SearchIcon,
  SpaceShuttleIcon,
} from "@patternfly/react-icons";
import classNames from "classnames";
import type {
  ComponentType,
  MouseEvent as ReactMouseEvent,
  ReactNode,
  VoidFunctionComponent,
} from "react";
import { NotFoundIcon } from "../../images";
import "./EmptyState.css";

export enum EmptyStateVariant {
  GettingStarted = "GettingStarted",
  NoAccess = "NoAccess",
  NoResult = "NoResult",
  NoItems = "NoItems",
  UnexpectedError = "UnexpectedError",
  PageNotFound = "PageNotFound",
}

export type EmptyStateProps = {
  titleProps?: Omit<TitleProps, "children" | "headingLevel"> & {
    headingLevel?: TitleProps["headingLevel"];
  };
  emptyStateProps?: Omit<PFEmptyStateProps, "children" | "variant"> & {
    variant?: EmptyStateVariant | PFEmptyStateVariant;
  };
  emptyStateIconProps?: EmptyStateIconProps;
  emptyStateBodyProps?: Omit<EmptyStateBodyProps, "children"> & {
    body?: string | ReactNode;
  };
  buttonProps?: Omit<ButtonProps, "children"> & {
    title?: string;
    onClick?: (event: ReactMouseEvent<HTMLButtonElement, MouseEvent>) => void;
    "data-testid"?: string;
  };
  children?: ReactNode;
};

export const EmptyState: VoidFunctionComponent<EmptyStateProps> = ({
  titleProps,
  buttonProps,
  emptyStateIconProps,
  emptyStateProps,
  emptyStateBodyProps,
  children,
}) => {
  const {
    variant: buttonVariant = ButtonVariant.primary,
    onClick,
    ...restButtonProps
  } = buttonProps || {};
  const { title, ...restTitleProps } = titleProps || {};
  const { body, ...restEmptyStateBodyProps } = emptyStateBodyProps || {};
  const {
    variant: masEmptyStateVariant = EmptyStateVariant.GettingStarted,
    className,
    ...restEmptyStateProps
  } = emptyStateProps || {};

  const NotFoundImg = () => (
    <img src={NotFoundIcon} alt="Not found page" width="128px" />
  );

  const getVariantConfig = () => {
    let variantConfig: {
      variant: PFEmptyStateVariant;
      icon?:
        | ComponentType<unknown>
        | typeof SpaceShuttleIcon
        | typeof LockIcon
        | typeof PlusCircleIcon
        | typeof SearchIcon
        | typeof ExclamationCircleIcon
        | typeof NotFoundImg;
      titleSize: TitleProps["size"];
      headingLevel: TitleProps["headingLevel"];
    };
    switch (masEmptyStateVariant) {
      case EmptyStateVariant.GettingStarted:
        variantConfig = {
          variant: PFEmptyStateVariant.xl,
          icon: SpaceShuttleIcon,
          titleSize: TitleSizes["4xl"],
          headingLevel: "h1",
        };
        break;
      case EmptyStateVariant.NoAccess:
        variantConfig = {
          variant: PFEmptyStateVariant.large,
          icon: LockIcon,
          titleSize: TitleSizes.xl,
          headingLevel: "h2",
        };
        break;
      case EmptyStateVariant.NoItems:
        variantConfig = {
          variant: PFEmptyStateVariant.large,
          icon: PlusCircleIcon,
          titleSize: TitleSizes.xl,
          headingLevel: "h2",
        };
        break;
      case EmptyStateVariant.NoResult:
        variantConfig = {
          variant: PFEmptyStateVariant.large,
          icon: SearchIcon,
          titleSize: TitleSizes.lg,
          headingLevel: "h2",
        };
        break;
      case EmptyStateVariant.UnexpectedError:
        variantConfig = {
          variant: PFEmptyStateVariant.full,
          icon: ExclamationCircleIcon,
          titleSize: TitleSizes.lg,
          headingLevel: "h1",
        };
        break;
      case EmptyStateVariant.PageNotFound:
        variantConfig = {
          variant: PFEmptyStateVariant.full,
          icon: NotFoundImg,
          titleSize: TitleSizes.lg,
          headingLevel: "h1",
        };
        break;
      default:
        variantConfig = {
          variant: masEmptyStateVariant || PFEmptyStateVariant.full,
          icon: emptyStateIconProps?.icon,
          titleSize: titleProps?.size || TitleSizes.lg,
          headingLevel: titleProps?.headingLevel || "h2",
        };
        break;
    }
    return variantConfig;
  };

  const { variant, icon, titleSize, headingLevel } = getVariantConfig();

  return (
    <PFEmptyState
      variant={variant}
      className={classNames("pf-u-pt-2xl pf-u-pt-3xl-on-md", className)}
      {...restEmptyStateProps}
    >
      <EmptyStateIcon icon={icon} {...emptyStateIconProps} />
      {title && (
        <Title headingLevel={headingLevel} size={titleSize} {...restTitleProps}>
          {title}
        </Title>
      )}
      {body && (
        <EmptyStateBody {...restEmptyStateBodyProps}>{body}</EmptyStateBody>
      )}
      {buttonProps?.title && (
        <Button variant={buttonVariant} onClick={onClick} {...restButtonProps}>
          {buttonProps?.title}
        </Button>
      )}
      {children}
    </PFEmptyState>
  );
};
