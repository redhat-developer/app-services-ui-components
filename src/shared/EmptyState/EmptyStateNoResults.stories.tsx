import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { EmptyStateNoResults } from "./EmptyStateNoResults";

export default {
  component: EmptyStateNoResults,
  args: {},
} as ComponentMeta<typeof EmptyStateNoResults>;

const Template: ComponentStory<typeof EmptyStateNoResults> = (args) => (
  <EmptyStateNoResults {...args} />
);

export const SearchResultNoFound = Template.bind({});
SearchResultNoFound.args = {};
