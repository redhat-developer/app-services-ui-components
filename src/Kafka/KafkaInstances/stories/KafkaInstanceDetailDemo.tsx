import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
