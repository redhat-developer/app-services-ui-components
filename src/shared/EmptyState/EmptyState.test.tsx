import { composeStories } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import * as stories from "./EmptyState.stories";

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName!,
  Story,
]);

// Batch snapshot testing
test.each(testCases)("Renders %s story", (_storyName, Story) => {
  const tree = render(<Story />);
  expect(tree.baseElement).toMatchSnapshot();
});
