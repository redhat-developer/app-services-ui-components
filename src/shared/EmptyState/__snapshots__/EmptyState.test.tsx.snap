// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-m-xl pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 640 512"
            width="1em"
          >
            <path
              d="M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z"
            />
          </svg>
          <h1
            class="pf-c-title pf-m-4xl"
            data-ouia-component-id="OUIA-Generated-Title-1"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Default Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders Getting Started story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-m-xl pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 640 512"
            width="1em"
          >
            <path
              d="M592.604 208.244C559.735 192.836 515.777 184 472 184H186.327c-4.952-6.555-10.585-11.978-16.72-16H376C229.157 137.747 219.403 32 96.003 32H96v128H80V32c-26.51 0-48 28.654-48 64v64c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v16c-23.197 0-32 10.032-32 24v40c0 13.983 8.819 24 32 24v64c0 35.346 21.49 64 48 64V352h16v128h.003c123.4 0 133.154-105.747 279.997-136H169.606c6.135-4.022 11.768-9.445 16.72-16H472c43.777 0 87.735-8.836 120.604-24.244C622.282 289.845 640 271.992 640 256s-17.718-33.845-47.396-47.756zM488 296a8 8 0 0 1-8-8v-64a8 8 0 0 1 8-8c31.909 0 31.942 80 0 80z"
            />
          </svg>
          <h1
            class="pf-c-title pf-m-4xl"
            data-ouia-component-id="OUIA-Generated-Title-2"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Getting Started Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders No Access story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-m-lg pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
            />
          </svg>
          <h1
            class="pf-c-title pf-m-xl"
            data-ouia-component-id="OUIA-Generated-Title-3"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            No Access Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders No Items story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-m-lg pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            />
          </svg>
          <h1
            class="pf-c-title pf-m-xl"
            data-ouia-component-id="OUIA-Generated-Title-4"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            No Items Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders No Result story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-m-lg pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <h1
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-5"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            No Result Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders Page Not Found story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <img
            alt="Not found page"
            src="dummyimage"
            width="128px"
          />
          <h1
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-6"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Page Not Found Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Renders Unexpected Error story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-c-empty-state pf-u-pt-2xl pf-u-pt-3xl-on-md"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            class="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
          <h1
            class="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-7"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Unexpected Error Empty State
          </h1>
          <div
            class="pf-c-empty-state__body"
          >
            
    Lorem ipsum odor amet, consectetuer adipiscing elit. Ac purus in massa egestas mollis varius
    dignissim elementum. Mollis tincidunt mattis hendrerit dolor eros enim, nisi ligula ornare.
    Hendrerit parturient habitant pharetra rutrum gravida porttitor eros feugiat. Mollis elit
    sodales taciti duis praesent id. Consequat urna vitae morbi nunc congue.
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
